app/src/main/java/com/fcaputo/parkingapp/mvp/view/SettingsView.kt

```diff
@@ -0,0 +1,76 @@
+package com.fcaputo.parkingapp.mvp.view
+
+import android.app.Activity
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.fcaputo.parkingapp.R
+import com.fcaputo.parkingapp.databinding.ActivitySettingsBinding
+import com.fcaputo.parkingapp.mvp.contract.SettingsContract
+import com.fcaputo.parkingapp.mvp.view.base.ActivityView
+import com.fcaputo.parkingapp.utils.Constants
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class SettingsView(activity: Activity) : ActivityView(activity),SettingsContract.View {
+    private var binding: ActivitySettingsBinding = ActivitySettingsBinding.inflate(activity.layoutInflater)
+
+    init {
+        activity.setContentView(binding.root)
+    }
+
+    override fun onSaveButton(onClick: () -> Unit) {
+        binding.saveButton.setOnClickListener{
+            if(fieldsAreComplete()) {
+                onClick()
+            } else {
+                showErrorMessage(ValidationErrorType.INCOMPLETE_FIELD)
+            }
+        }
+    }
+
+    private fun fieldsAreComplete(): Boolean = listOf(binding.tInputParkingSize).all{ it.editText?.text?.isNotEmpty() ?: false }
+
+    override fun clear() { binding.tInputParkingSize.editText?.text?.clear() }
+
+    override fun getParkingSize(): Int = binding.tInputParkingSize.editText?.text.toString().toInt()
+
+    override fun showParkingSize(size: Int) {
+        val sizeString = activity?.resources?.getString(R.string.settings_current_size) + (if (size == Constants.SIZE_NOT_SET_INT) " -" else " $size")
```

Comment:
You could use a string resource with parameters for this!
https://medium.com/@sourav.bh/best-practices-for-using-string-resources-in-android-c947fdd1606a

Reply:
Like this?
<img width="282" alt="image" src="https://github.com/CFedericoDavid/g-reskilling-android-parkingApp/assets/13878290/ed7e69ae-fbc5-41fe-9042-a4aa19412cdc">

<img width="587" alt="image" src="https://github.com/CFedericoDavid/g-reskilling-android-parkingApp/assets/13878290/129b1104-a608-40ee-ae07-66986c17d4e9">

Reply:
Exactly! 🚀

Reply:
Done